whc-shared/src/main/java/com/ibm/whc/deid/shared/pojo/config/masking/BinningMaskingProviderConfig.java

```diff
@@ -17,99 +17,169 @@
+		type = MaskingProviderType.BINNING;
+	}
+
+	public int getBinSize() {
+		return binSize;
+	}
+
+	public void setBinSize(int binSize) {
+		this.binSize = binSize;
+	}
+
+	public String getFormat() {
+		return format;
+	}
+
+	public void setFormat(String format) {
+		this.format = format == null ? DEFAULT_FORMAT : format;
+	}
+
+	public int getStartValue() {
+		return startValue;
+	}
+
+	public void setStartValue(int startValue) {
+		this.startValue = startValue;
+	}
+
+	public boolean isUseStartValue() {
+		return useStartValue;
+	}
+
+	public void setUseStartValue(boolean useStartValue) {
+		this.useStartValue = useStartValue;
+	}
+
+	public boolean isUseSingleBucketOverThreshold() {
+		return useSingleBucketOverThreshold;
+	}
+
+	public void setUseSingleBucketOverThreshold(boolean useSingleBucketOverThreshold) {
+		this.useSingleBucketOverThreshold = useSingleBucketOverThreshold;
+	}
+
+	public double getSingleBucketOverThresholdValue() {
+		return singleBucketOverThresholdValue;
+	}
+
+	public void setSingleBucketOverThresholdValue(double singleBucketOverThresholdValue) {
+		this.singleBucketOverThresholdValue = singleBucketOverThresholdValue;
+	}
+
+	public String getSingleBucketOverThresholdReplacement() {
+		return singleBucketOverThresholdReplacement;
+	}
+
+	public void setSingleBucketOverThresholdReplacement(String singleBucketOverThresholdReplacement) {
+		this.singleBucketOverThresholdReplacement = singleBucketOverThresholdReplacement;
+	}
+
+	public boolean isUseSingleBucketUnderThreshold() {
+		return useSingleBucketUnderThreshold;
+	}
+
+	public void setUseSingleBucketUnderThreshold(boolean useSingleBucketUnderThreshold) {
+		this.useSingleBucketUnderThreshold = useSingleBucketUnderThreshold;
+	}
+
+	public double getSingleBucketUnderThresholdValue() {
+		return singleBucketUnderThresholdValue;
+	}
+
+	public void setSingleBucketUnderThresholdValue(double singleBucketUnderThresholdValue) {
+		this.singleBucketUnderThresholdValue = singleBucketUnderThresholdValue;
+	}
+
+	public String getSingleBucketUnderThresholdReplacement() {
+		return singleBucketUnderThresholdReplacement;
+	}
+
+	public void setSingleBucketUnderThresholdReplacement(String singleBucketUnderThresholdReplacement) {
+		this.singleBucketUnderThresholdReplacement = singleBucketUnderThresholdReplacement;
+	}
+	
+	@Override
+	public void validate(DeidMaskingConfig maskingConfig) throws InvalidMaskingConfigurationException {
+		super.validate(maskingConfig);
+		if (binSize < 1) {
+			throw new InvalidMaskingConfigurationException("`binSize` must be greater than 0");
+		}
+		 if (singleBucketOverThresholdReplacement == null) {
+		      throw new InvalidMaskingConfigurationException("`singleBucketOverThresholdReplacement` must not be null");
+		 }
+		 if (singleBucketUnderThresholdReplacement == null) {
+		      throw new InvalidMaskingConfigurationException("`singleBucketUnderThresholdReplacement` must not be null");
+		 }
+		 if (singleBucketOverThresholdValue < 1) {

```

Comment:
The over threshold can be any value.  Since a primitive double can't be null, this check isn't needed.